modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceRecreateApi.java

```diff
@@ -0,0 +1,212 @@
+import org.activiti.engine.runtime.ProcessInstance;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.api.config.AuditApiRequest;
+import org.eclipse.xpanse.modules.database.service.DeployServiceEntity;
+import org.eclipse.xpanse.modules.database.servicetemplate.ServiceTemplateEntity;
+import org.eclipse.xpanse.modules.deployment.DeployServiceEntityHandler;
+import org.eclipse.xpanse.modules.deployment.recreate.RecreateService;
+import org.eclipse.xpanse.modules.deployment.recreate.consts.RecreateConstants;
+import org.eclipse.xpanse.modules.models.service.deploy.exceptions.BillingModeNotSupported;
+import org.eclipse.xpanse.modules.models.service.deploy.exceptions.EulaNotAccepted;
+import org.eclipse.xpanse.modules.models.service.deploy.exceptions.InvalidServiceStateException;
+import org.eclipse.xpanse.modules.models.service.deploy.exceptions.ServiceLockedException;
+import org.eclipse.xpanse.modules.models.service.enums.ServiceDeploymentState;
+import org.eclipse.xpanse.modules.models.workflow.recreate.RecreateRequest;
+import org.eclipse.xpanse.modules.models.workflow.recreate.enums.RecreateStatus;
+import org.eclipse.xpanse.modules.models.workflow.recreate.view.ServiceRecreateDetails;
+import org.eclipse.xpanse.modules.security.UserServiceHelper;
+import org.eclipse.xpanse.modules.workflow.utils.WorkflowUtils;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * REST interface methods for Service Recreate.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_ADMIN, ROLE_USER})
+@ConditionalOnProperty(name = "enable.agent.api.only", havingValue = "false", matchIfMissing = true)
+public class ServiceRecreateApi {
+
+    @Resource
+    private DeployServiceEntityHandler deployServiceEntityHandler;
+
+    @Resource
+    private UserServiceHelper userServiceHelper;
+
+    @Resource
+    private WorkflowUtils workflowUtils;
+
+    @Resource
+    private RecreateService recreateService;
+
+    /**
+     * Create a job to recreate the deployed service.
+     *
+     * @return response
+     */
+    @Tag(name = "Recreate", description = "APIs to manage the service recreate.")
```

Comment:
we do not need a new tag here. The tag and description must be same as 

@Tag(name = "Service", description = "APIs to manage the services")